File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    int len = s.size();
    int weight = 'A' - 'a';
    string answer = "";
    for(int i = 0; i < len; i ++){
        if(i == 0 || isspace(s[i-1])){
            if(islower(s[i])) answer += s[i] + weight;
            else answer += s[i];
        }
        else{
            if(isupper(s[i])) answer += s[i] - weight;
            else answer += s[i];
        }
    }

    return answer;
}

File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    flag = 1
    for now in s:
        if now == ' ':
            answer += ' '
            flag = 1
            
        else:
            if flag:
                answer += now.upper()
                flag = 0
            else:
                answer += now.lower()
    
    return answer